src/isomorphic-fetch.js

```diff
@@ -1,9 +1,16 @@
+/* globals require */
+
 import global from './metal/global';
 import isNodeLikeEnvironment from './metal/is-node-like-environment';
 
 const fetch = global.fetch;
 
 if (!fetch && isNodeLikeEnvironment()) {
-  global.fetch = global.require('node-fetch');
+  /* global.require is only available in the REPL 
+   * and Babel fails when you use require() explicitly
+   */
```

Comment:
You can repeatably cause this bug under the amd processor. The commonjs one seems fine. There's also a long outstanding bug which I couldn't find documenting it only on the amd processor.